certbot/main.py

```diff
@@ -388,7 +388,15 @@ def _tos_cb(regr):
 
 
 def _delete_if_appropriate(config): # pylint: disable=too-many-locals,too-many-branches
-    """Does the user want to delete their now-revoked certs?"""
+    """Does the user want to delete their now-revoked certs? If run in non-interactive mode,
```

Comment:
Doesn't it also happen automatically even if we're not in non-interactive mode?

Reply:
Yes, I should re-phrase this. It was written a very, very long time ago.

Reply:
You know, are we sure this isn't the correct description, and the code is wrong? When did we decide to always just delete it? Didn't we decide to give an option, and have the default choice (ie when you just hit enter, implying that that's what happens in the non-interactive case) be to delete it?

Reply:
ah, found it https://github.com/certbot/certbot/pull/4202/files/6e07c333bf84a0869c82be2a5962cb2a3fb484c2#r115631846

Reply:
Given everything else about how the code currently is, what do you think?

Reply:
Er, I'm a bit confused by your previous comment vs most recent ones. To be clear, which do you think is best?

Ultimately, I'm OK with either option, but this whole time I have been leaning towards at least giving users an option of saving their certs (as it might be useful for security auditing/retrospectives), even though I implemented deleting by default (because I thought that was the conclusion we came to as a result of some design discussions). It would only take a few hours to change it back to the previous behaviour.

Reply:
I really don't remember the previous design discussions we've had at this point, and the comment I put above was all I could find written down on the topic, which isn't much.

We also might want to worry about key pinning; people might want to keep their keys around even if they've revoked one cert.

If you're leaning towards asking, sounds good to me. It should really be only the one question, with default set to delete it, I think.

Reply:
Ah, I was referencing an in-person meeting. And good point re: key pinning. I'll modify the behaviour/add the question in soon :)